internal/nix/shell.go

```diff
@@ -371,11 +364,11 @@ func (s *Shell) writeDevboxShellrc() (path string, err error) {
 		return "", fmt.Errorf("create temp dir for shell init file: %v", err)
 	}
 
-	// This is a best-effort to include the user's existing shellrc. If we
-	// can't read it, then just omit it from the devbox shellrc.
-	userShellrc, err := os.ReadFile(s.userShellrcPath)
-	if err != nil {
-		userShellrc = []byte{}
+	// This is a best-effort to include the user's existing shellrc.
+	userShellrc := []byte{}
+	if s.userShellrcPath != "" {
+		// If we can't read it, then just omit it from the devbox shellrc.
+		userShellrc, _ = os.ReadFile(s.userShellrcPath)
```

Comment:
would prefer also keeping the older code that checks that `err != nil` and only then writing the value to `userShellrc`